gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/AssignmentColumnHeaderPanel.java

```diff
@@ -369,4 +377,24 @@ private Integer calculateCurrentCategorizedSortOrder(final Long assignmentId) {
 
 		return order;
 	}
+
+	private boolean canUserEditAssignment(GbRole role, Assignment assignment) {
+		boolean canEdit = false;
+		if (role == GbRole.INSTRUCTOR) {
+			canEdit = true;
+		} else {
+			List<PermissionDefinition> permissions = this.businessService.getPermissionsForUser(
```

Comment:
I feel bad hitting the service for this for each header cell.  Perhaps we could do a sweep of all the panels and centralise any businessService data somewhere in the page to reduce service calls?  I don't think this is the only guilty panel.